web/datapipeline/hosts_projector.go

```diff
@@ -99,7 +99,7 @@ func storeHost(db *gorm.DB, host entities.Host, updateColumns ...string) error {
 		Columns: []clause.Column{
 			{Name: "agent_id"},
 		},
-		DoUpdates: clause.AssignmentColumns(append(updateColumns, "updated_at")),
+		DoUpdates: clause.AssignmentColumns(append(updateColumns, "agent_id", "updated_at")),
```

Comment:
Is there any chance of change of the `agent_id` ?

Reply:
@fabriziosestito can help me better here :sweat_smile: 
I don't really remember what it fixes, sorry :man_facepalming:

Reply:
Oh no problem, I see how this can be confusing.
So basically it just ensures that `agent_id` column is always written, prior to this commit we were just creating Hosts entities when the host discovery event was received. Since we cannot guarantee the order of the events received, this ensures that if we receive for example a cloud discovery event first, we create the host as well.
If the host already exists, it should not change the `agent_id` because of the `on conflict` clause above.